src/api/rpc_api/author.rs

```diff
@@ -60,51 +63,58 @@ where
 	/// extrinsic progress.
 	fn submit_and_watch_extrinsic(
 		&self,
-		xthex_prefixed: &str,
+		encoded_extrinsic: Vec<u8>,
 	) -> Result<TransactionSubscriptionFor<Client, Hash>>;
 
 	/// Submit an extrinsic and watch in until the desired status is reached,
-	/// if no error is encountered previously. This method is blocking.
+	/// if no error is encountered previously.
+	// This method is blocking.
 	fn submit_and_watch_extrinsic_until(
 		&self,
-		xthex_prefixed: &str,
+		encoded_extrinsic: Vec<u8>,
 		watch_until: XtStatus,
-	) -> Result<Option<Hash>>;
+	) -> Result<ExtrinsicReport<Hash>>;
 }
 
 impl<Signer, Client, Params, Runtime> SubmitAndWatch<Client, Runtime::Hash>
 	for Api<Signer, Client, Params, Runtime>
 where
 	Client: Subscribe,
 	Params: ExtrinsicParams<Runtime::Index, Runtime::Hash>,
-	Runtime: FrameSystemConfig,
+	Runtime: FrameSystemConfig + GetRuntimeBlockType,
+	Runtime::RuntimeBlock: BlockTrait + DeserializeOwned,
```

Comment:
Why is the `RuntimeBlock` traitbound needed?

Reply:
Ah, that's not yet necessary, my bad. Good catch!